server/app/api/repo/repo_service.py

```diff
@@ -18,6 +19,22 @@ def __init__(self, session: Session):
         self.user_repo = UserRepo(session)
         self.org_repo = OrganizationRepo(session)
         self.access_control_service = AccessControlService(session)
+        self.team_repo = TeamRepo(session)
+
+    # -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=- #
+    # Utility methods. Many of these already exist as methods of other services, but we can't
+    # use them because of circular dependencies (FastAPI does not support it).
+    
+    def _update_repo_attribute(self, repo: Repository, dto: RepositoryDescUpdateDTO | RepositoryVisibilityUpdateDTO) -> Repository:
+        if dto is None:
+            raise AccessDeniedException(f"Repository {repo.id} cannot be updated with a None value")
+        elif isinstance(dto, RepositoryDescUpdateDTO):
```

Comment:
Instead of `AccessDeniedException`, how about something like BadRequest or InvalidPayload or something like that?